### ISSUE TITLE
Random failing of TestBug297635

### ISSUE THREAD

**Issue by fedejeanne** [eclipse-platform/eclipse.platform#460]
The test `TestBug297635.test2` fails randomly.

One such failures can be seen in https://github.com/eclipse-platform/eclipse.platform/actions/runs/4991112285/jobs/8937471805.

Here's the stack trace:

```
junit.framework.AssertionFailedError
	at junit.framework.Assert.fail(Assert.java:55)
	at junit.framework.Assert.assertTrue(Assert.java:22)
	at junit.framework.Assert.assertNotNull(Assert.java:256)
	at junit.framework.Assert.assertNotNull(Assert.java:248)
	at junit.framework.TestCase.assertNotNull(TestCase.java:391)
	at org.eclipse.core.tests.resources.session.TestBug297635.assertStateTrees(TestBug297635.java:169)
	at org.eclipse.core.tests.resources.session.TestBug297635.test2(TestBug297635.java:131)
...
```

**Comment by HeikoKlare** [comment]
The failing test was obviously not (properly) fixed by #465, as it still fails randomly:
https://github.com/eclipse-platform/eclipse.platform/actions/runs/5369959755/jobs/9742278023

**Comment by HeikoKlare** [comment]
In my opinion, the real issue with this test is that is in an overall bad state. It validates internal states using reflection, even though that is not what the test is supposed to test. What the test is actually supposed to do (according to the bug it is a regression test for) is to ensure that saving snapshots clears the save state trees (by calling `forgetSavedTree`). So actually the test should check whether that method is properly called when saving snapshots rather than whether some internal state changed. Whether or not there has been some state stored before saving a snapshot or not does not matter for the test. The `forgetSavedTree` method is responsible for doing that (no matter how, when or why it was called).

So in my opinion a proper test for the bug would look much easier:
- Create a mock/spy workspace
- Create a `SaveManager` for that workspace
- Make a snapshot save on the `SaveManager` instance
- Validate that `forgetSavedTree` was called on the workspace

Everything else (whether internal states are properly cleared by the `forgetSavedTree` call and the like) should not be part of the test and is verified separately.
